client/src/components/ActivityShow.js

```diff
@@ -26,9 +27,38 @@ const ActivityShow = (props) => {
     }
   };
 
+  const postReview = async (newReview) => {
+    try {
+      const response = await fetch(`/api/v1/activities/${activityId}/reviews`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newReview)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const errorBody = await response.json()
+          const newErrors = translateServerErrors(errorBody.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+      } else {
+        const reviewData = await response.json()
+        setErrors([])
+        setActivity({ ...activity }, activity.reviews.push(reviewData.review))
```

Comment:
hmm, remember that `push` is a mutating method and should not be used with state. 

If we want to add the review to the nested array in state:
```js
setActivity({
  ...activity,
  reviews: [...activity.reviews, reviewData.review]
})
```
should do the trick